trinity/nodes/base.py

```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+import asyncio
+from pathlib import PurePath
+from threading import Thread
+from typing import Type
+
+from evm.chains.base import BaseChain
+from evm.db.header import BaseHeaderDB
+from p2p.service import (
+    BaseService,
+    EmptyService,
+)
+from trinity.rpc.main import (
+    RPCServer,
+)
+from trinity.rpc.ipc import (
+    IPCServer,
+)
+
+
+class Node(BaseService):
```

Comment:
There seems to be some overlap between this and `p2p.Server`, and we were planning to extend the latter to take a `ChainConfig` instance and then instantiate the appropriate syncer, etc. Do you think we'd be able to combine them somehow?

Reply:
I think we're headed in the same general direction. In general, `p2p.Server` will only be one of many services that the node needs to manage (like the json-rpc server, etc). So my instinct is that this Node service should take the `ChainConfig` and start up `p2p.Server` with that info, to handle all the syncing needs.